dbt/models/default/default.vw_pin_sale.sql

```diff
@@ -249,93 +239,179 @@ sales_val AS (
+        COALESCE(uq_sales.num_parcels_sale, md_sales.num_parcels_sale)
+            AS num_parcels_sale_coalesced,
+        COALESCE(uq_sales.buyer_name, md_sales.buyer_name)
+            AS buyer_name_coalesced,
+        COALESCE(uq_sales.sale_type, NULL) AS sale_type_coalesced,
+        uq_sales.max_price,
+        uq_sales.bad_doc_no,
+        CASE WHEN uq_sales.doc_no IS NOT NULL THEN 'iasworld' ELSE 'mydec' END
+            AS source,
+        md_sales.mydec_deed_type,
+        md_sales.sale_filter_ptax_flag,
+        md_sales.mydec_property_advertised,
+        md_sales.mydec_is_installment_contract_fulfilled,
+        md_sales.mydec_is_sale_between_related_individuals_or_corporate_affiliates, --noqa
+        md_sales.mydec_is_transfer_of_less_than_100_percent_interest,
+        md_sales.mydec_is_court_ordered_sale,
+        md_sales.mydec_is_sale_in_lieu_of_foreclosure,
+        md_sales.mydec_is_condemnation,
+        md_sales.mydec_is_short_sale,
+        md_sales.mydec_is_bank_reo_real_estate_owned,
+        md_sales.mydec_is_auction_sale,
+        md_sales.mydec_is_seller_buyer_a_relocation_company,
+        md_sales.mydec_is_seller_buyer_a_financial_institution_or_government_agency, --noqa
+        md_sales.mydec_is_buyer_a_real_estate_investment_trust,
+        md_sales.mydec_is_buyer_a_pension_fund,
+        md_sales.mydec_is_buyer_an_adjacent_property_owner,
+        md_sales.mydec_is_buyer_exercising_an_option_to_purchase,
+        md_sales.mydec_is_simultaneous_trade_of_property,
+        md_sales.mydec_is_sale_leaseback,
+        md_sales.mydec_is_homestead_exemption,
+        md_sales.mydec_homestead_exemption_general_alternative,
+        md_sales.mydec_homestead_exemption_senior_citizens,
+        md_sales.mydec_homestead_exemption_senior_citizens_assessment_freeze
+    FROM unique_sales AS uq_sales
+    FULL OUTER JOIN mydec_sales AS md_sales ON uq_sales.doc_no = md_sales.doc_no
+    LEFT JOIN town_class AS tc
+        ON COALESCE(uq_sales.pin, md_sales.pin) = tc.parid
+        AND COALESCE(uq_sales.year, md_sales.year) = tc.taxyr
+),
+
+-- Handle various filters 
+combined_sales AS (
+    SELECT
+        cte_s.*,
+
+        -- Calculate 'sale_filter_same_sale_within_365' using DATE_DIFF
+        CASE
+            WHEN LAG(cte_s.sale_date_coalesced) OVER (
+                    PARTITION BY cte_s.pin_coalesced, cte_s.sale_price_coalesced
+                    ORDER BY cte_s.sale_date_coalesced ASC
+                ) IS NOT NULL
+                THEN
+                DATE_DIFF(
+                    'day',
+                    LAG(cte_s.sale_date_coalesced) OVER (
+                        PARTITION BY --noqa
+                            cte_s.pin_coalesced, cte_s.sale_price_coalesced
+                        ORDER BY cte_s.sale_date_coalesced ASC
+                    ),
+                    cte_s.sale_date_coalesced
+                ) <= 365
+            ELSE FALSE
+        END AS sale_filter_same_sale_within_365,
```

Comment:
I think this is a big improvement. Let me know what you think @wrridgeway

Reply:
Agreed. Let's do some testing - let's make sure that if we limit our source to iasWorld the new view looks like the old view, and that invoking filters still returns the same numbers of rows.